src/Libraries/PythonNodeModels/PythonNode.cs

```diff
@@ -228,6 +242,30 @@ protected override bool UpdateValueCore(UpdateValueParams updateValueParams)
             return base.UpdateValueCore(updateValueParams);
         }
 
+        /// <summary>
+        /// Updates the Script property of the node.
+        /// NOTE: This is a temporary method used during the Python 2 to Python 3 transistion period,
+        /// it will be removed when the transistion period is over.
+        /// </summary>
+        /// <param name="newCode">The new migrated code</param>
+        public void MigrateCode(string newCode)
+        {        
+            var e = new PythonCodeMigrationEventArgs(Script, newCode);
+            OnCodeMigrated(e);
+            Script = newCode;
+        }
+
+        /// <summary>
+        /// Fires when the Script content is migrated to Python 3.
+        /// NOTE: This is a temporary event used during the Python 2 to Python 3 transistion period,
```

Comment:
let's see if we can decorate these temporary methods with `[Obsolete]` or some attribute like that, will show up as warnings for devs using them, so they know they will be removed in the future

Reply:
Building on this, can we ask the Dynamo team if they want to track tasks to remove these methods?